test_scripts/API/Restructuring_OnResetTimeout/002_BC_OnResetTimeout_Success_flow_without_resetPeriod.lua

```diff
@@ -0,0 +1,60 @@
+------------------------------------------------------------------------------------------------------------------------
+-- Proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0189-Restructuring-OnResetTimeout.md
+------------------------------------------------------------------------------------------------------------------------
+-- Description: Check SDL is able to reset timeout for Mobile app response to default value
+--  by receiving 'OnResetTimeout(nil)' notification from HMI
+------------------------------------------------------------------------------------------------------------------------
+-- Preconditions:
+-- 1) Default SDL timeout is 10s (defined in .INI by 'DefaultTimeout' parameter)
+--
+-- In case:
+-- 1) App sends applicable RPC
+-- 2) SDL transfers this request to HMI
+-- 3) HMI sends 'BC.OnResetTimeout' notification to SDL without 'resetPeriod' parameter within the delay of 5s
+-- after receiving request from SDL
+-- 4) HMI sends response after 11s
+-- SDL does:
+--  - wait for the response from HMI within reset period
+--  - once response received proceed with it successfully and transfer it to Mobile app
+------------------------------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local common = require('test_scripts/API/Restructuring_OnResetTimeout/common_OnResetTimeout')
+
+--[[ Apps configuration ]]
+common.getConfigAppParams(1).appHMIType = { "REMOTE_CONTROL" }
+common.getConfigAppParams(2).appHMIType = { "REMOTE_CONTROL" }
+
+--[[ Local Variables ]]
+local paramsForRespFunction = {
+  respTime = common.defaultTimeout + 1000,
+  notificationTime = 5000,
+  resetPeriod = nil
+}
+
+local RespParams = { success = true, resultCode = "SUCCESS" }
+
+--[[ Scenario ]]
+common.Title("Preconditions")
+common.Step("Clean environment", common.preconditions)
+common.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+common.Step("App_1 registration", common.registerAppWOPTU)
+common.Step("App_2 registration", common.registerAppWOPTU, { 2 })
+common.Step("App_1 activation", common.activateApp)
+common.Step("Set RA mode: ASK_DRIVER", common.defineRAMode, { true, "ASK_DRIVER" })
+common.Step("Create InteractionChoiceSet id 100", common.createInteractionChoiceSet, { 100 })
+common.Step("Create InteractionChoiceSet id 200", common.createInteractionChoiceSet, { 200 })
+common.Step("Add AddSubMenu", common.addSubMenu)
+
+common.Title("Test")
+for _, rpc in pairs(common.rpcsArray) do
+  common.Step("Send " .. rpc , common.rpcs[rpc],
+    { common.defaultTimeout + 2000, 6000, common.responseWithOnResetTimeout,
```

Comment:
Since we decided with https://github.com/smartdevicelink/sdl_core/pull/3726#issuecomment-888553791 (the addition of the `DefaultTimeoutCompensation` param) that SDL would compensate for the time required to process a request, I think the padding for the OnResetTimeout notifications should be removed. 
```suggestion
    { common.defaultTimeout, 6000, common.responseWithOnResetTimeout,
```

Reply:
@ShobhitAd Please notice 1st argument for each `common.rpcs.<RPC>` function is `pExpTimoutForMobResp`. This is the max timeout for ATF to wait a mobile response from SDL. In scenario defined by this script, response from SDL to mobile app would come after 11 sec. (`common.defaultTimeout + 1000`). So ATF require some additional buffer to be able to receive this response. We usually define 1 sec. for such buffer. In our case it would be 12 sec. (`common.defaultTimeout + 2000`).